metadata/add_plot_boundaries.Rmd

```diff
@@ -0,0 +1,142 @@
+
+# Set up
+
+Load in the `sf` and `RPostgreSQL` packages.
+
+```{r load-pack, message = FALSE}
+
+library(sf)
+library(RPostgreSQL)
+
+```
+
+Create a connection to the BETY test database.
+
+```{r create-dbcon}
+
+dbcon <- dbConnect(PostgreSQL(),
+                   host = 'localhost',
+                   user = 'bety',
+                   port = 5433)
+
+```
+
+# Adding plot boundaries using `sf` package
+
+## Read in shapefiles using `st_read`
+
+MAC Season 8 shapefiles can be read in as a `simple features object` using `st_read`. These shapefiles are saved in the `MAC_season8` folder of this directory. 
+
+A `sf` object is a dataframe where rows are features (points, lines, polygons) and columns are attributes of the features.
+
+```{r read-shape, results = 'hide'}
+
+s8_bounds <- st_read('MAC_season8/S8_one_row_polys.shp')
+
+```
+
+If you need to convert the coordinate reference system, you can use the `st_transform` function. In this example, we will be converting the EPSG from `32612` to `4326`.
+
+```{r transform-crs}
+
+s8_bounds <- st_transform(s8_bounds, 4326)
+
+```
+
+## Write a table to BETYdb using `st_write`
+
+The `st_write` function can be used to write a new table to BETYdb containing the contents of the `sf` object. The new table will be named `s8_bounds` and will be dropped after it is used to insert new data into the `sites` table.
+
+```{r write-table}
+
+st_write(s8_bounds, dbcon)
+
+```
+
+# Updating and inserting new records using `RPostgreSQL` package
+
+The MAC Season 8 shapefiles did not inlude a `sitename` attribute. We will need to add this field to the `s8_bounds` table and update its values.
+
+The `dbSendStatement` function from the `RPostgreSQL` package will be used to execute SQL statements. 
+
+The following statements will be executed:
+
+`alter table s8_bounds add column sitename VARCHAR;`
+
+`update s8_bounds set sitename = concat('MAC Field Scanner Season 8 Range ', f_range, ' Column ', f_column, ' ', row);`
+
+```{r create-update-sitename, results = 'hide'}
+
+# create sitename field in s8_bounds table
+create_sitename <- dbSendStatement(dbcon,
+                                   "alter table s8_bounds add column sitename VARCHAR;")
+dbClearResult(create_sitename)
+
+# populate sitename values to be concatenation of multiple fields from table
+update_sitename <- dbSendStatement(dbcon,
+                                   paste0("update s8_bounds ",
+                                          "set sitename = concat(",
+                                          "'MAC Field Scanner Season 8 Range ', ",
+                                          "f_range, ",
+                                          "' Column ', ",
+                                          "f_column, ",
+                                          "' ', ",
+                                          "row);"))
+dbClearResult(update_sitename)
+
+```
+
+## Add new records to sites table using subset of s8_bounds
```

Comment:
Should we have calls to st_isvalid() here to verify geometry? I think it'd be good to at least mention that here so that people can be sure their shapes are good. Might need a shape trouble shooting section?

Reply:
Yes, I think it is a good idea to verify the geometries. 

I have updated tutorial to check geometries using ST_IsValid and make updates to invalid geometries using ST_MakeValid. Do you know of any better ways to do this?